src/main/java/org/reaktivity/nukleus/socks/internal/stream/SocksServerFactory.java

```diff
@@ -649,10 +627,12 @@ private void onApplicationBegin(
             OctetsFW extension = begin.extension();
             SocksBeginExFW socksBeginEx = extension.get(socksBeginExRO::wrap);
 
-            String address = socksBeginEx.address().asString();
+            SocksAddressFW replyAddress = socksAddressRO.wrap(socksBeginEx.address().buffer(),
+                socksBeginEx.address().offset(),
+                socksBeginEx.address().limit());
```

Comment:
Please hoist `socksBeginEx.address()` into its own variable to avoid calling same method 3 times.

Reply:
done